src/adapter/auth-oso-rebac/src/oso_resource_service_inmem.rs

```diff
@@ -0,0 +1,222 @@
+// Copyright Kamu Data, Inc. and contributors. All rights reserved.
+//
+// Use of this software is governed by the Business Source License
+// included in the LICENSE file.
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0.
+
+use std::collections::HashMap;
+
+use dill::*;
+use internal_error::InternalError;
+use kamu_core::{
+    DatasetLifecycleMessage,
+    DatasetLifecycleMessageCreated,
+    DatasetLifecycleMessageDeleted,
+    DatasetVisibility,
+    MESSAGE_PRODUCER_KAMU_CORE_DATASET_SERVICE,
+};
+use messaging_outbox::{
+    MessageConsumer,
+    MessageConsumerMeta,
+    MessageConsumerT,
+    MessageDeliveryMechanism,
+};
+use opendatafabric as odf;
+use tokio::sync::{RwLock, RwLockReadGuard};
+
+use crate::{
+    DatasetResource,
+    UserActor,
+    MESSAGE_CONSUMER_KAMU_AUTH_OSO_REBAC_OSO_RESOURCE_SERVICE,
+};
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+type EntityId = String;
+
+#[derive(Debug, Default)]
+struct State {
+    user_actor_map: HashMap<EntityId, UserActor>,
+    dataset_resource_map: HashMap<EntityId, DatasetResource>,
+}
+
+////////////////////////////////////////////////////////////////////////////////////////////////////////////////////////
+
+pub struct OsoResourceServiceInMem {
+    state: RwLock<State>,
+}
```

Comment:
Is this really a service? Looks like an in-mem repository to me.

Reply:
`OsoResourceServiceInMem` has been renamed `OsoResourceServiceImpl` and its implementation has become more service-like